src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/TemplateBuilder.cs

```diff
@@ -108,6 +96,40 @@ public IHtmlContent Build()
             // though _model may have been reset to null. Otherwise we might lose track of the model type /property.
             viewData.ModelExplorer = _modelExplorer.GetExplorerForModel(_model);
 
+            var formattedModelValue = viewData.Model;
+            if (_model == null && _readOnly)
+            {
+                formattedModelValue = _metadata.NullDisplayText;
+            }
+            else if (viewData.ModelMetadata.IsEnum)
+            {
+                // Cover the case where the model is an enum and we want the string value of it
+                var modelEnum = viewData.Model as Enum;
+                if (modelEnum != null)
+                {
+                    var value = modelEnum.ToString("d");
+                    var enumGrouped = viewData.ModelMetadata.EnumGroupedDisplayNamesAndValues;
+                    Debug.Assert(enumGrouped != null);
+                    foreach (var kvp in enumGrouped)
+                    {
+                        if (kvp.Value == value)
+                        {
+                            // Creates a ModelExplorer with the same Metadata except that the Model is a string instead of an Enum
+                            formattedModelValue = kvp.Key.Name;
+                            break;
+                        }
+                    }
+                }
+            }
+
+            var formatString = _readOnly ?
+                viewData.ModelMetadata.DisplayFormatString
+                : viewData.ModelMetadata.EditFormatString;
```

Comment:
This looks really messy. Stick w/ the operators at the end of the lines unless you feel strongly (that's more common in this repo). But at least place them consistently w/in this one expression!